modules/tapadIdSystem.js

```diff
@@ -0,0 +1,78 @@
+import { uspDataHandler } from '../src/adapterManager.js';
+import { submodule } from '../src/hook.js';
+import { getStorageManager } from '../src/storageManager.js';
+import * as ajax from '../src/ajax.js'
+import * as utils from '../src/utils.js';
+
+export const storage = getStorageManager();
+const cookiesMaxAge = 7 * 24 * 60 * 60 * 1000; // 60 days
+export const defaultExpirationString = new Date(utils.timestamp() + cookiesMaxAge).toString();
+export const pastDateString = new Date(0).toString();
+export const tapadCookieKey = 'tapad_id';
+export const graphUrl = 'https://realtime-graph-access-zxvhwknfeq-uc.a.run.app/v1/graph';
+
+function saveOnAllStorages(key, value, expiration = defaultExpirationString) {
+  storage.setCookie(key, typeof value === 'object' ? JSON.stringify(value) : value, expiration);
+  storage.setDataInLocalStorage(key, JSON.stringify(value));
+}
+
+function deleteFromAllStorages(key) {
+  storage.setCookie(key, undefined, pastDateString);
+  storage.removeDataFromLocalStorage(key);
+}
```

Comment:
In your documentation example (https://github.com/prebid/prebid.github.io/pull/2606/files) you have the publisher set the `storage` object, yet you are manually storing data here. If `storage` is set, then the user id module will handle storing and retrieving the id; if you want to do it manually then publishers cannot set `storage` or you will end up with inconsistencies. 

Also, if you are going to manage storage on your own, you need to set an expiration item in the local storage and manually handle throwing it out when reading the data if it's expired. 

i would suggest just using the built-in functionality of the user id module to read/write the cached id value.